libtransact/src/contract/archive/mod.rs

```diff
@@ -14,3 +14,586 @@
+//!   - The contract `name`
+//!   - The contract `version`
+//!   - The contract's `input` addresses
+//!   - The contract's `output` addresses
+//!
+//! # Examples
+//!
+//! ```no_run
+//! use transact::contract::archive::{default_scar_paths, SmartContractArchive};
+//!
+//! let default_paths = default_scar_paths();
+//! let archive = SmartContractArchive::from_scar_file("xo", "0.1", &default_paths[0])
+//!     .expect("failed to load .scar file");
+//! ```
+
+mod error;
+
+use std::env::{split_paths, var_os};
+use std::fs::File;
+use std::io::Read;
+use std::path::{Path, PathBuf};
+
+use bzip2::read::BzDecoder;
+use glob::glob;
+use semver::{Version, VersionReq};
+use tar::Archive;
+
+pub use error::Error;
+
+const DEFAULT_SCAR_PATH: &str = "/usr/share/scar";
+const SCAR_PATH_ENV_VAR: &str = "SCAR_PATH";
+
+/// Load default location(s) for .scar files.
+pub fn default_scar_paths() -> Vec<PathBuf> {
+    match var_os(SCAR_PATH_ENV_VAR) {
+        Some(paths) => split_paths(&paths).collect(),
+        None => vec![PathBuf::from(DEFAULT_SCAR_PATH)],
+    }
+}
+
+/// The definition of a smart contract, including the contract's associated metadata and the bytes
+/// of the smart contract itself.
+pub struct SmartContractArchive {
+    /// The raw bytes of the WASM smart contract.
+    pub contract: Vec<u8>,
+    /// The metadata associated with this smart contract.
+    pub metadata: SmartContractMetadata,
+}
+
+impl SmartContractArchive {
+    /// Attempt to load a `SmartContractArchive` from a .scar file in the given `path`.
+    ///
+    /// The .scar file to load will be named `<name>_<version>.scar`, where `<name>` is the name
+    /// specified for this method and `<version>` is a valid semver string.
+    ///
+    /// The `version` argument for this method must be a valid semver requirement string. The
+    /// latest contract (based on semver requirement rules) that matches the given name and meets
+    /// the version requirements will be loaded.
+    ///
+    /// For more info on semver: https://semver.org/
+    ///
+    /// # Example
+    ///
+    /// If there are two files `/usr/share/scar/xo_1.2.3.scar` and `/usr/share/scar/xo_1.2.4`, you
+    /// can load v1.2.4 with:
+    ///
+    /// ```no_run
+    /// use transact::contract::archive::SmartContractArchive;
+    ///
+    /// let archive = SmartContractArchive::from_scar_file("xo", "1.2", "/usr/share/scar")
+    ///     .expect("failed to load .scar file");
+    /// assert_eq!(&archive.metadata.version, "1.2.4");
+    /// ```
+    ///
+    /// To load v1.2.3 explicitly, specify the exact version:
+    ///
+    /// ```no_run
+    /// use transact::contract::archive::SmartContractArchive;
+    ///
+    /// let archive = SmartContractArchive::from_scar_file("xo", "1.2.3", "/usr/share/scar")
+    ///     .expect("failed to load .scar file");
+    /// assert_eq!(&archive.metadata.version, "1.2.3");
+    /// ```
+    pub fn from_scar_file<P: AsRef<Path>>(
```

Comment:
load_scar_file() would also make sense here, since it is loading it from the filesystem.